include/common/reduce_pattern.h

```diff
@@ -65,7 +65,7 @@ class reduce_t {
   \brief Check if a reduction can be performed.
   */
```

Comment:
We should add a doxygen precondtion

`\pre items.size() < static_cast<int>(std::numeric_limits<int>::max())`